File: Sample/.Net/PublicApiDemo/Common/PlexePublicAPIDemo.Common/CalculationService.cs
using Newtonsoft.Json;
using PlexePublicAPIDemo.Common.Model;
using RestSharp;
using System;
using System.Collections.Generic;
using System.Net;
using System.Text;

namespace PlexePublicAPIDemo.Common
{
    public class CalculationService
    {
        private readonly string BASE_URI = "https://apidemo.plexe.co/";
        private readonly string APPLICATION_URI = "api/Calculation/";

        public CalculateLOCSliderResponse CalculateLOCSlider(string key, string applicationId)
        {
            CalculateLOCSliderResponse calculateLOCSliderResponse = new CalculateLOCSliderResponse();
            try
            {                
                var client = new RestClient(BASE_URI + APPLICATION_URI + "application-loc-calculation-slider?applicationId=" + applicationId + "");
                var request = new RestRequest();
                request.AddHeader("Content-Type", "application/json");
                request.AddHeader("Accept", "application/json");
                request.AddHeader("Authorization", "Bearer " + key);
                request.Method = Method.Post;
                SliderRequest sliderRequest = new SliderRequest()
                {
                     Amount = 23
                };
                var body = JsonConvert.SerializeObject(sliderRequest);
                request.AddParameter("application/json", body, ParameterType.RequestBody);
                RestResponse response = client.Execute(request);
                calculateLOCSliderResponse = JsonConvert.DeserializeObject<CalculateLOCSliderResponse>(response.Content);
                Console.WriteLine(response.Content);
                return calculateLOCSliderResponse;
            }
            catch (Exception ex)
            {
                return null;
            }
        }
        public ApplicationLocCalculation ApplicationLocCalculation(string key, string applicationId)
        {
            ApplicationLocCalculation applicationLocCalculationResponse = new ApplicationLocCalculation();
            try
            {
                var client = new RestClient(BASE_URI + APPLICATION_URI + "application-loc-calculation?applicationId=" + applicationId + "");
                var request = new RestRequest();
                request.AddHeader("Content-Type", "application/json");
                request.AddHeader("Accept", "application/json");
                request.AddHeader("Authorization", "Bearer " + key);
                request.Method = Method.Post;
                CalculationRequest calculationRequest = new CalculationRequest()
                {
                    Amount = 23,
                    FixedRepaymentCalculation = true,
                    PercentOfIncome = 2,
                    Terms = 22
                };
                var body = JsonConvert.SerializeObject(calculationRequest);
                request.AddParameter("application/json", body, ParameterType.RequestBody);
                RestResponse response = client.Execute(request);
                Console.WriteLine(response.Content);
                applicationLocCalculationResponse = JsonConvert.DeserializeObject<ApplicationLocCalculation>(response.Content);
                return applicationLocCalculationResponse;
            }
            catch (Exception ex)
            {
                return null;
            }
        }
    }    
}

File: Sample/Php/PublicApiDemo/services/CalculationService.php
<?php

require_once 'vendor/autoload.php';
use GuzzleHttp\Client;
use GuzzleHttp\Exception\RequestException;
use GuzzleHttp\Psr7\Request;
  
class CalculationService
{
    private $BASE_URI = "https://apidemo.plexe.co/";
    private $APPLICATION_URI ="api/Calculation/";

    function StatusCodeHandling($e)
    {
        if ($e -> getResponse() -> getStatusCode() == '400')
        {
            return 'BAD REQUEST';
            //return 'BAD REQUEST';
            //$this->prepare_access_token();
        }
        else if ($e -> getResponse() -> getStatusCode() == '422')
        {
            $response = json_decode($e -> getResponse()->getBody(true)->getContents());
            return $response;
        }
        else if ($e -> getResponse() -> getStatusCode() == '500')
        {
            $response = json_decode($e ->getResponse()->getBody(true)->getContents());
            return $response;
        }
        else if ($e -> getResponse() -> getStatusCode() == '401')
        {
            $response = json_decode($e ->getResponse()->getBody(true)->getContents());
            return $response;
        }
        else if ($e->getResponse()->getStatusCode() == '403')
        {
            $response = json_decode($e->getResponse()->getBody(true)->getContents());
            return $response;
        }
        else
        {
            $response = json_decode($e->getResponse()->getBody(true)->getContents());
            return $response;
        }       
    }

    public function CalculateLOCSlider($key, $applicationId)
    {
        try
        {                
            $client = new Client();
            $headers = [
              'Content-Type' => 'application/json',
              'Accept' => 'application/json',
              'Authorization' => 'Bearer '.$key
            ];
            $body = '{
              "amount": 34
            }';
            $request = new Request('POST', $this->BASE_URI.$this->APPLICATION_URI.'application-loc-calculation-slider?applicationId='.$applicationId, $headers, $body);
            $res = $client->sendAsync($request)->wait();
            $result = $res->getBody();
            $decoded = json_decode($result);
            return $decoded;
        }
        catch (RequestException $e)
        {
            return 'BAD_REQUEST';
        }
    }

    public function ApplicationLocCalculation($key, $applicationId)
    {
        try
        {
            $client = new Client();
            $headers = [
              'Content-Type' => 'application/json',
              'Accept' => 'application/json',
              'Authorization' => 'Bearer '.$key
            ];
            $body = '{
              "amount": 23,
              "percentOfIncome": 22,
              "terms": 6,
              "fixedRepaymentCalculation": true
            }';
            $request = new Request('POST', $this->BASE_URI.$this->APPLICATION_URI.'application-loc-calculation?applicationId='.$applicationId, $headers, $body);
            $res = $client->sendAsync($request)->wait();
            $result = $res->getBody();
            $decoded = json_decode($result);
            return $decoded;

        }
        catch (RequestException $e)
        {
            return 'BAD_REQUEST';
        }
    }
}
?>